packages/apps/dev-wallet/src/hooks/useTheme.ts

```diff
@@ -0,0 +1,143 @@
+import { darkThemeClass } from '@kadena/react-ui/styles';
+import { useCallback, useEffect, useState } from 'react';
+const MEDIA = '(prefers-color-scheme: dark)';
+const storageKey = 'theme';
+const isServer = typeof window === 'undefined';
+const defaultTheme = 'system';
+
+export type ITheme = 'dark' | 'light' | 'system';
+interface IUseThemeProps {
+  /**
+   * lockedTheme will lock the theme for the whole application.
+   * It will overwrite the localstorage and even the switching of themes in your system will not update the theme
+   * IMPORTANT: using the useTheme twice in a application can lead to unexpected race conditions
+   * You would need to lock the theme on all the uses of this hook with the same ITheme
+   */
+  lockedTheme?: ITheme;
+  /**
+   * overwriteTheme can be used when you want 1 particular component to have a fixed theme
+   * IMPORTANT: this will also overwrite the lockedTheme for that component
+   */
+  overwriteTheme?: ITheme;
```

Comment:
As far as I can see, the theme is applied at the application level. How can you set it at the component level?